examples/tv-app/tv-common/clusters/messages/MessagesManager.cpp

```diff
@@ -18,62 +18,143 @@
 #include "MessagesManager.h"
 
 #include <app-common/zap-generated/attributes/Accessors.h>
+#include <vector>
 
 using namespace std;
+using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters::Messages;
 using Message = chip::app::Clusters::Messages::Structs::MessageStruct::Type;
 
 // Commands
-void MessagesManager::HandlePresentMessagesRequest(
-    const chip::ByteSpan & messageId, const MessagePriorityEnum & priority,
-    const chip::BitMask<MessageControlBitmap> & messageControl, const chip::app::DataModel::Nullable<uint32_t> & startTime,
-    const chip::app::DataModel::Nullable<uint16_t> & duration, const chip::CharSpan & messageText,
-    const chip::Optional<chip::app::DataModel::DecodableList<MessageResponseOption>> & responses)
+void MessagesManager::HandlePresentMessagesRequest(const ByteSpan & messageId, const MessagePriorityEnum & priority,
+                                                   const BitMask<MessageControlBitmap> & messageControl,
+                                                   const DataModel::Nullable<uint32_t> & startTime,
+                                                   const DataModel::Nullable<uint16_t> & duration, const CharSpan & messageText,
+                                                   const Optional<DataModel::DecodableList<MessageResponseOption>> & responses)
 {
-    Message message{
-        // TODO: Enable id
-        chip::ByteSpan(), priority, messageControl, startTime, duration,
-        // TODO: Enable text
-        chip::CharSpan()
-        // TODO: Convert responses to Optional<chip::app::DataModel::List<const
-        // chip::app::Clusters::Messages::Structs::MessageResponseOptionStruct::Type>> message.responses = responses;
-    };
-
-    mMessages.push_back(message);
+    std::unique_ptr<char[]> messageTextBufferPtr;
+    std::unique_ptr<uint8_t[]> messageIdBufferPtr;
+
+    uint8_t * messageIdBuffer = nullptr;
+    // see MessagesManager.h "Note on memory management" for allocation/free rules
+    char * messageTextBuffer = new char[messageText.size() + 1];
```

Comment:
Why 2 separate values that you convert to a std::string below anyway? Why not create a std::string to start with (it would malloc and not reuse pointers anyway)?

Or otherwise you could `ScopedMemoryBuffer txt; VerifyOrExit(txt.Alloc(messageText.size() + 1); ...` and not have so many mixed variables.

Reply:
moved memory management to CachedMessage and CachedMessageOption